OPHD/XmlSerializer.h

```diff
@@ -27,6 +28,19 @@ T stringToEnum(const std::unordered_map<std::string, T>& table, std::string valu
 }
 
 
+template<typename T>
+T stringToEnum(const std::map<std::string, T>& table, std::string value)
+{
+	auto it = table.find(value);
+	if (it != table.end())
+	{
+		return it->second;
+	}
+
+	throw std::runtime_error("Unable to parse enum with value of " + value);
+}
```

Comment:
Slightly irrelevant observation, but this basically implements [`std::map::at`](https://en.cppreference.com/w/cpp/container/map/at), but with a better exception message. I don't see any way to meaningfully improve this though, nor use `std::map::at` and still get a better exception message.